registry.go

```diff
@@ -0,0 +1,79 @@
+package manet
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	ma "github.com/jbenet/go-multiaddr"
+)
+
+type AddrParser func(a net.Addr) (ma.Multiaddr, error)
+type MaddrParser func(ma ma.Multiaddr) (net.Addr, error)
+
+var maddrParsers map[string]MaddrParser
+var addrParsers map[string]AddrParser
+var addrParsersLock sync.Mutex
+
+type AddressSpec struct {
+	// NetNames is an array of strings that may be returned
+	// by net.Addr.Network() calls on addresses belonging to this type
+	NetNames []string
+
+	// Key is the string value for Multiaddr address keys
+	Key string
+
+	// ParseNetAddr parses a net.Addr belonging to this type into a multiaddr
+	ParseNetAddr AddrParser
+
+	// ConvertMultiaddr converts a multiaddr of this type back into a net.Addr
+	ConvertMultiaddr MaddrParser
+
+	// Protocol returns the multiaddr protocol struct for this type
+	Protocol ma.Protocol
+}
+
+func RegisterAddressType(a *AddressSpec) {
```

Comment:
Its tested indirectly. None of the tests would function if this function didnt work right :P